### ISSUE TITLE
[Cache] Memcached $defaultClientOptions override passed serializer

### ISSUE THREAD

**Issue by duronrulez** [symfony/symfony#33717]
**Symfony version(s) affected**: 3.4.* (but pretty sure all version have it)

**Description**  
When you want to change the serializer option for memcached, it gets overriden by the default options.

**How to reproduce**  
`MemcachedAdapter::createConnection($serversDsn, [\Memcached::OPT_SERIALIZER => \Memcached::SERIALIZER_IGBINARY]);`
// this creates the connections with the default serializer PHP from the MemcachedTrait $defaultClientOptions

**Possible Solution**  
instead of using this:

```php
    private static $defaultClientOptions = [
        'persistent_id' => null,
        'username' => null,
        'password' => null,
        'serializer' => 'php',
    ];
```

use this:

```php
    private static $defaultClientOptions = [
        'persistent_id' => null,
        'username' => null,
        'password' => null,
        \Memcached::OPT_SERIALIZER => \Memcached::SERIALIZER_PHP,
    ];
```

**Comment by nicolas-grekas** [comment]
Fixed in #33719

Meanwhile, you can use
`MemcachedAdapter::createConnection($serversDsn, ['serializer' => 'igbinary']);`

**Comment by duronrulez** [comment]
Thanks, yes thats what we've applied temporary.
